app/src/main/java/nextstep/payments/ui/card/CardViewModel.kt

```diff
@@ -0,0 +1,33 @@
+package nextstep.payments.ui.card
+
+import androidx.lifecycle.ViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import nextstep.payments.data.model.Card
+import nextstep.payments.data.repository.PaymentCardsRepository
+
+class CardViewModel(
+    private val repository: PaymentCardsRepository = PaymentCardsRepository
+): ViewModel() {
+
+    private val _creditCardUiState = MutableStateFlow<CreditCardUiState>(CreditCardUiState.Empty)
+    val creditCardUiState: StateFlow<CreditCardUiState> = _creditCardUiState.asStateFlow()
+
+    fun fetchCards() {
+        val cards = repository.cards
+        if (cards.isEmpty()) {
+            _creditCardUiState.value = CreditCardUiState.Empty
+        } else if (cards.size == 1) {
+            _creditCardUiState.value = CreditCardUiState.One(cards[0])
+        } else {
+            _creditCardUiState.value = CreditCardUiState.Many(cards)
+        }
+    }
+}
+
+sealed interface CreditCardUiState {
+    data object Empty : CreditCardUiState
+    data class One(val card: Card) : CreditCardUiState
+    data class Many(val cards: List<Card>): CreditCardUiState
+}
```

Comment:
아래와 같이 CreditCardUiState에 분기문조차 캡슐화해버릴 수도 있을 것 같아요!
용호님은 이러한 구조에 대해 어떻게 생각하시나요?

```suggestion
    fun fetchCards() {
        val cards = repository.cards
        _creditCardUiState.value = CreditCardUiState.from(cards)
    }
}

sealed interface CreditCardUiState {
    data object Empty : CreditCardUiState
    data class One(val card: Card) : CreditCardUiState
    data class Many(val cards: List<Card>): CreditCardUiState
    
    companion object {
        fun from(cards: List<Card>) {
	    return if (cards.isEmpty()) {
	        CreditCardUiState.Empty
	    } else if (cards.size == 1) {
	        CreditCardUiState.One(cards.first())
	    } else {
	        CreditCardUiState.Many(cards)
	    }
        }
    }
}
```

Reply:
이런 식으로 캡슐화를 하니깐 훨씬 깔끔하네요..!